sklearn/linear_model/_base.py

```diff
@@ -717,7 +717,8 @@ def rmatvec(b):
                 self.coef_ = np.vstack([out[0] for out in outs])
                 self._residues = np.vstack([out[3] for out in outs])
         else:
-            self.coef_, self._residues, self.rank_, self.singular_ = linalg.lstsq(X, y)
+            self.coef_, self._residues, self.rank_, self.singular_ =\
+                 linalg.lstsq(X, y, check_finite=False)
```

Comment:
In this case `X` is returned by `_preprocess_data`, which can lead to finite values, even with finite input:

```python
from sklearn.linear_model import LinearRegression
import numpy as np

f_max = np.finfo(np.float64).max
X = np.full((10, 10), 0.9 * f_max)
y = np.full(10, 0)

print(np.isfinite(X).all())
# True
print(np.isfinite(y).all())
# True

X, y, X_offset, y_offset, X_scale = LinearRegression._preprocess_data(X, y, fit_intercept=True)

print(np.isfinite(X).all())
# False
```